docs/changelog/3090.feature.rst

```diff
@@ -0,0 +1 @@
+Add support for setting build backend ``config_settings`` in the configuration file - by :user:`gaborbernat`.
```

Comment:
@nschloe did you forget to use your own username here?

Reply:
Not really. I made _some_ contributions here, but Gabor finished it all up.